tests/test_validator.py

```diff
@@ -61,17 +67,25 @@ def test_invalid_ids(self):
         self.assertFalse(
             validate("naw.aberavon.2019-01-01")
         )  # should have a subtype
-        self.assertFalse(
-            validate("gla.r.barnet-and-camden.2016-05-05")
-        )  # invalid subtype
-        self.assertFalse(validate("naw.x.2019-01-01"))  # invalid subtype
         self.assertFalse(
             validate("ref.croydon.by.2021-10-07")
         )  # there is no such thing as a by-referendum
+
+        # invalid subtypes
+        self.assertFalse(validate("gla.r.barnet-and-camden.2016-05-05"))
+        self.assertFalse(validate("naw.x.2019-01-01"))
+        self.assertFalse(validate("senedd.x.2021-05-06"))
+        self.assertFalse(validate("senedd.x.mid-and-west-wales.2021-05-06"))
+
         # too many clauses
         self.assertFalse(
             validate("naw.r.mid-and-west-wales.something-else.2016-05-05")
         )
+        self.assertTrue(
+            validate(
+                "senedd.pen-y-bont-bro-morgannwg.something-else.2026-05-07"
```

Comment:
This seems to be a separate change? In this case I think the only thing that goes between `division` and `date` is an optional `by`.
I guess the other way of reading the id is as a division called `something-else` with a subtype of `pen-y-bont-bro-morgannwg`. But then shouldn't this be invalid, because that wouldn't be a valid subtype.